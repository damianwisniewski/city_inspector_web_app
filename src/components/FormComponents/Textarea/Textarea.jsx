import React from 'react'
import PropTypes from 'prop-types'

const Textarea = ({ id, placeholder, label, floatingLabel, ...rest }) => (
	<div className='input-wrapper'>
		<textarea
			{...rest}
			data-animate={floatingLabel}
			className='input-wrapper__field'
			id={id}
			placeholder={placeholder}
		/>
		<label className='input-wrapper__label' htmlFor={id}>
			{label}
		</label>
	</div>
)

Textarea.propTypes = {
	label: '',
}

Textarea.propTypes = {
	id: PropTypes.string.isRequired,
	placeholder: PropTypes.string.isRequired,
	label: PropTypes.string,
	floatingLabel: PropTypes.bool.isRequired,
}

export default Textarea
