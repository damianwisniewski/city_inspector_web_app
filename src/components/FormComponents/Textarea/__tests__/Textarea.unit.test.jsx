import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import Textarea from '../Textarea'

configure({ adapter: new Adapter() })

describe('<Textarea />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let TextareaElement
	const TextareaContent = 'mockTextarea'
	const id = 'mockId'
	const placeholder = 'mock placeholder'

	test('should render correctly Textarea, with floatingLabel value equal true', () => {
		TextareaElement = shallow(
			<Textarea floatingLabel placeholder={placeholder} id={id} label={TextareaContent} />,
		)
		expect(toJson(TextareaElement)).toMatchSnapshot()
	})

	test('should render correctly Textarea, with floatingLabel value equal false', () => {
		TextareaElement = shallow(
			<Textarea labelType='basic' placeholder={placeholder} id={id} label={TextareaContent} />,
		)
		expect(toJson(TextareaElement)).toMatchSnapshot()
	})
})
