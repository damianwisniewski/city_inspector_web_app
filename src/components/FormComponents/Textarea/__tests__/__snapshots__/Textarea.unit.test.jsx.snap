// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> should render correctly Textarea, with floatingLabel value equal false 1`] = `
<div
  className="input-wrapper input-wrapper--basic"
>
  <textarea
    className="input-wrapper__field"
    data-animate={false}
    id="mockId"
    placeholder="mock placeholder"
  />
  <label
    className="input-wrapper__label input-wrapper__label--basic"
    htmlFor="mockId"
  >
    mockTextarea
  </label>
</div>
`;

exports[`<Textarea /> should render correctly Textarea, with floatingLabel value equal true 1`] = `
<div
  className="input-wrapper input-wrapper--undefined"
>
  <textarea
    className="input-wrapper__field"
    data-animate={false}
    floatingLabel={true}
    id="mockId"
    placeholder="mock placeholder"
  />
  <label
    className="input-wrapper__label input-wrapper__label--undefined"
    htmlFor="mockId"
  >
    mockTextarea
  </label>
</div>
`;
